File: src/main/java/dev/retrotv/crypto/owe/hash/sha/SHA3512.java
package dev.retrotv.crypto.owe.hash.sha;

import dev.retrotv.crypto.owe.hash.Hash;
import dev.retrotv.utils.EncodeUtil;
import dev.retrotv.utils.MessageDigestEncodeUtil;

import static dev.retrotv.enums.HashAlgorithm.SHA3512;

public class SHA3512 extends Hash {

    @Override
    public String hash(byte[] data) {
        return EncodeUtil.binaryToHex(MessageDigestEncodeUtil.encode(SHA3512, data));
    }
}

File: src/main/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA3512.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.hash.HashAlgorithm
import dev.retrotv.data.enums.EncodeFormat
import dev.retrotv.data.utils.toHexString
import dev.retrotv.enums.Algorithm.Hash.SHA3512
import dev.retrotv.utils.digest
import dev.retrotv.utils.encode

/**
 * SHA3-512 알고리즘으로 암호화 하기 위한 [HashAlgorithm] 추상 클래스의 구현체 입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
class SHA3512 : HashAlgorithm() {
    override fun hash(data: ByteArray): String {
        return toHexString(digest(SHA3512, data))
    }

    override fun hash(data: ByteArray, encodeFormat: EncodeFormat): String {
        return encode(encodeFormat, digest(SHA3512, data))
    }
}
